modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -203,22 +225,115 @@ public CompletableFuture<Void> startAsync(ComponentContext componentContext) {
                         inBusyLock(busyLock, () -> onCreateZone(parameters).thenApply((ignored) -> false))
         );
 
-        return nullCompletedFuture();
+        return allOf(processZonesOnStart, processAssignmentsFuture);
+    }
+
+    private CompletableFuture<Void> processZonesOnStart(long recoveryRevision, @Nullable HybridTimestamp lwm) {
+        int earliestCatalogVersion = catalogMgr.activeCatalogVersion(hybridTimestampToLong(lwm));
+        int latestCatalogVersion = catalogMgr.latestCatalogVersion();
```

Comment:
Is it guaranteed that latestCatalogVersion will be in sync with metaStorageMgr.recoveryFinishedFuture()? I mean that catalogManager start is async now.

Reply:
todo added https://issues.apache.org/jira/browse/IGNITE-22679